### PR TITLE
Don't linkify code blocks

### PR DESCRIPTION
**Pull request by robintown**
Type: defect

Closes https://github.com/vector-im/element-web/issues/9613.

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Don't linkify code blocks ([\#7859](https://github.com/matrix-org/matrix-react-sdk/pull/7859)). Fixes vector-im/element-web#9613. Contributed by @robintown.

Preview: https://pr7859--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

**Comment by t3chguy** [comment]
Hmm, it seems that github linkifies things in code blocks - so not sure whether this is the right path unfortunately

image

```
foo
https://github.com/matrix-org/matrix-react-sdk/pull/7859
https://github.com
github.com
bar
```

Maybe we should still linkify explicit URLs but skip implicit (github.com) ones. Github seems to never linkify implicit schema-less URLs.

**Comment by robintown** [comment]
Good point, I'll see if there's an option for that…

**Comment by robintown** [comment]
> Maybe we should still linkify explicit URLs but skip implicit (github.com) ones. Github seems to never linkify implicit schema-less URLs.

It looks like the only way to do this would be to set a class name on implicit URLs that causes them to be styled like normal text when inside code blocks (super hacky :stuck_out_tongue:)

**Comment by t3chguy** [comment]
> edit: oh, but of course if we were to skip implicit URLs in general, that would be doable

I doubt this would fly by product unfortunately.

I think the className based approach is not entirely insane, with CSS we could strip styling and pointer-actions, wonder if that'd prevent highlighting though

**Comment by robintown** [comment]
I spoke too soon, apparently the `className` option in linkifyjs only lets you decide what classes to add based on the href and not the text itself, so you can't tell whether the text contains a scheme or not :( I think I'll just leave this as-is, then.

File: test/components/views/messages/TextualBody-test.js [hunk 1]
```diff
@@ -196,6 +196,33 @@ describe("<TextualBody />", () => {
...
+            expect(wrapper.text()).toBe("Visit https://matrix.org/\n1https://matrix.org/\n\n");
+            const content = wrapper.find(".mx_EventTile_body");
+            expect(content.html()).toBe('<span class="mx_EventTile_body markdown-body" dir="auto">' +
+                '<p>Visit <code>https://matrix.org/</code></p>\n' +
+                '<div class="mx_EventTile_pre_container">' +
+                '<pre class="mx_EventTile_collapsedCodeBlock">' +
+                '<span class="mx_EventTile_lineNumbers"><span>1</span></span>' +
+                '<code>https://matrix.org/\n</code><span></span></pre>' +
+                '<span class="mx_EventTile_button mx_EventTile_copyButton ">' +
+                '</span></div>\n</span>');
```
Comment (t3chguy) [inline test/components/views/messages/TextualBody-test.js:223]: Can we use the snapshot utility here? Keeps tests a bit more readable IMO

**Review by t3chguy — APPROVED** [review]
LGTM code-wise

**Review by marcusmueller — APPROVED** [review]

**Comment by daniellekirkwood** [comment]
Screenshot 2022-05-02 at 10 15 50

this is what i see - is this the expected behaviour as discussed above?

**Comment by robintown** [comment]
@daniellekirkwood Yes, that's expected

**Comment by daniellekirkwood** [comment]
GReat, then product approves 👍

**Review by daniellekirkwood — APPROVED** [review]

**Pull request merged**